rust/worker/src/segment/record_segment.rs

```diff
@@ -831,50 +832,19 @@ impl RecordSegmentReader<'_> {
     /// embedding id
     #[allow(dead_code)]
     pub(crate) async fn get_all_data(&self) -> Result<Vec<DataRecord>, Box<dyn ChromaError>> {
-        let mut data = Vec::new();
-        let max_size = self.user_id_to_id.count().await?;
-        for i in 0..max_size {
-            let res = self.user_id_to_id.get_at_index(i).await;
-            match res {
-                Ok((_, _, offset_id)) => {
-                    if let Some(data_record) = self.id_to_data.get("", offset_id).await? {
-                        data.push(data_record);
-                    } else {
-                        return Err(
-                            Box::new(RecordSegmentReaderCreationError::DataRecordNotFound(
-                                offset_id,
-                            )) as _,
-                        );
-                    }
-                }
-                Err(e) => {
-                    tracing::error!(
-                        "[GetAllData] Error getting data record for index {:?}: {:?}",
-                        i,
-                        e
-                    );
-                    return Err(e);
-                }
-            }
-        }
-        Ok(data)
+        self.id_to_data
+            .get_range(""..="", ..)
+            .await
+            .map(|vec| vec.into_iter().map(|(_, data)| data).collect())
     }
 
-    pub(crate) async fn get_offset_id_at_index(
-        &self,
-        index: usize,
-    ) -> Result<u32, Box<dyn ChromaError>> {
-        match self.id_to_user_id.get_at_index(index).await {
-            Ok((_, oid, _)) => Ok(oid),
-            Err(e) => {
-                tracing::error!(
-                    "[GetAllData] Error getting offset id for index {}: {}",
-                    index,
-                    e
-                );
-                Err(e)
-            }
-        }
+    pub(crate) fn get_offset_stream<'me>(
```

Comment:
Add a comment on the contract of this method. It's not clear from the names

Reply:
Added ccomments